examples/custom_websocket_source/main.py

```diff
@@ -0,0 +1,111 @@
+import json
+import logging
+from typing import List
+
+from dateutil.parser import isoparse
+from websockets import ConnectionClosedOK
+from websockets.sync.client import connect
+
+from quixstreams import Application
+from quixstreams.sources import Source
+
+logger = logging.getLogger(__name__)
+
+
+class CoinbaseSource(Source):
+    def __init__(
+        self, name: str, url: str, product_ids: List[str], shutdown_timeout: float = 10
+    ):
+        """
+        An example implementation of the custom websocket source
+        that connects to Coinbase Websocket API
+        and streams data to an intermediate Kafka topic.
+
+        Coinbase API docs - https://docs.cdp.coinbase.com/exchange/docs/websocket-overview
+
+        :param name: a unique name of the source.
+            It is used as a part of the default topic name.
+        :param url: Coinbase WS API URL.
+        :param product_ids: a list of Product IDs to receive from the API
+            (e.g., "ETH-BTC").
+        :param shutdown_timeout: Time in seconds the application waits
+            for the source to gracefully shutdown.
+        """
+        super().__init__(name=name, shutdown_timeout=shutdown_timeout)
+        self._url = url
+        self._product_ids = product_ids
+
+    def run(self) -> None:
+        """
+        The main method of the source with the main logic.
+        """
+
+        # Open a websocket connection
+        ws_conn = connect(self._url)
+
+        # Subscribe to the updates for the given Product IDs
+        subscribe_payload = {
+            "type": "subscribe",
+            "channels": [
+                {"name": "ticker", "product_ids": self._product_ids},
+            ],
+        }
+        ws_conn.send(json.dumps(subscribe_payload))
+
+        # Start the reading loop
+        while self.running:
+            try:
+                # Receive a message from the websocket
+                msg = ws_conn.recv(timeout=1)
+            except TimeoutError:
+                continue
```

Comment:
I think we should poll the producer here

Reply:
Shouldn't it happen automatically on each `.produce()` call?

Reply:
It won't happen if you don't receive anything from coinbase

Reply:
Currently, there's no public method for `.poll()`

Reply:
I'm going to merge it because the source is working.
I created #511  to address the gaps in Producer polling in the future